packages/datadog-plugin-protobufjs/test/index.spec.js

```diff
@@ -0,0 +1,309 @@
+'use strict'
+
+const fs = require('fs')
+const { expect } = require('chai')
+const agent = require('../../dd-trace/test/plugins/agent')
+const path = require('path')
+const {
+  SCHEMA_DEFINITION,
+  SCHEMA_ID,
+  SCHEMA_NAME,
+  SCHEMA_OPERATION,
+  SCHEMA_WEIGHT,
+  SCHEMA_TYPE
+} = require('../../dd-trace/src/constants')
+const sinon = require('sinon')
+
+const schemas = JSON.parse(fs.readFileSync(path.join(__dirname, 'schemas/expected_schemas.json'), 'utf8'))
+const MESSAGE_SCHEMA_DEF = schemas.MESSAGE_SCHEMA_DEF
+const OTHER_MESSAGE_SCHEMA_DEF = schemas.OTHER_MESSAGE_SCHEMA_DEF
+const ALL_TYPES_MESSAGE_SCHEMA_DEF = schemas.ALL_TYPES_MESSAGE_SCHEMA_DEF
+
+const MESSAGE_SCHEMA_ID = '666607144722735562'
+const OTHER_MESSAGE_SCHEMA_ID = '2691489402935632768'
+const ALL_TYPES_MESSAGE_SCHEMA_ID = '15890948796193489151'
+
+function compareJson (expected, span) {
+  const actual = JSON.parse(span.context()._tags[SCHEMA_DEFINITION])
+  return JSON.stringify(actual) === JSON.stringify(expected)
+}
+
+describe('Plugin', () => {
+  describe('protobufjs', function () {
+    this.timeout(0)
+
+    let tracer
+    let protobuf
+    let dateNowStub
+    let mockTime
+
+    beforeEach(() => {
+      return agent.load('protobufjs')
+        .then(() => {
+          protobuf = require('protobufjs')
+          tracer = require('../../dd-trace')
+        })
+    })
+
+    before(() => {
+      mockTime = Date.now()
+      dateNowStub = sinon.stub(Date, 'now').callsFake(() => {
+        const returnValue = mockTime
+        mockTime += 50000 // Increment by 50000 ms to ensure each DSM schema is sampled
+        return returnValue
+      })
+    })
+
+    afterEach(() => {
+      return agent.close()
+    })
+
+    after(() => {
+      dateNowStub.restore()
+    })
+
+    // it('should serialize basic schema correctly', async () => {
```

Comment:
The test is essentially the same as the deserialization function, but not clear to me why the serialization function tests are commented out

Reply:
Ahhh, that mustve snuck in accidentally, anyways, these tests have been updated. I also added some abstraction to prevent duplicate code.